packages/wiki-docs/Migrating.md

```diff
@@ -0,0 +1,231 @@
+      "publisher": "SeaSketch",
+      "publishDate": "20231018",
+      "publishLink": ""
+    },
+    "precalc": false
+  }]
+```
+
+Global EEZ datasource  published by [global-datasources](https://github.com/seasketch/global-datasources) (with filters set to for Samoa EEZ)
+
+```json
+[{
+    "datasourceId": "global-eez-mr-v12",
+    "geo_type": "vector",
+    "formats": [
+      "fgb",
+      "json"
+    ],
+    "precalc": true,
+    "metadata": {
+      "name": "World EEZ v11",
+      "description": "World EEZ boundaries and disputed areas",
+      "version": "11.0",
+      "publisher": "Flanders Marine Institute (VLIZ)",
+      "publishDate": "2019118",
+      "publishLink": "https://marineregions.org/"
+    },
+    "idProperty": "GEONAME",
+    "nameProperty": "GEONAME",
+    "classKeys": [],
+    "url": "https://gp-global-datasources-datasets.s3.us-west-1.amazonaws.com/global-eez-mr-v12.fgb",
+    "propertyFilter": {
+      "property": "GEONAME",
+      "values": [
+        "Samoan Exclusive Economic Zone"
+      ]
+    },
+    "bboxFilter": [
+      -174.51139447157757,
+      -15.878383591829206,
+      -170.54265693017294,
+      -10.960825304544073
+    ]
+  }]
+```
+
+
+Finally, you need to add a `precalc` setting to all other datasources in your `datasources.json` file.  This property is required, and you will see validation errors when running any data commands or smoke tests.
+
+- Add `"precalc": true` to all datasources in `project/datasources.json` that metrics should be precalculated for.  This is typically limited to datasources that you need to precalc the overall summary metric of the datasource (total area, total feature count) so that you can report the % of the total that a sketch overlaps with.  Otherwise you don't need to precalc.
+- Set all other datasources to `"precalc": false`.  This includes global datasources or datasources that are only a source for `geography` features and otherwise aren't used in reports.  Setting these to true will at best just precalculate extra metrics that won't be used.  At worst it will try to fetch entire global datasources and fail at the task, because the necessary filters aren't in place.
+- Create a new file `project/precalc.json` populated with an empty array `[]`
+
+### Precalc
+
+Once you have your `geographies` and `datasources` properly configured, you're ready to run the `precalc:data` command.  Make sure you have an empty array in 
+
+```bash
+npm run precalc:data
+```
+
+This will precompute metrics for all combinations of geographies and datasources. It will also strip any existing `keyStats` properties from `datasources.json` and populate `precalc.json`.
+
+### Geoprocessing functions
+
+- Update `clipToGeography` function, to allow geographies to use external datasources. Copy the following file in the gp library to your project space to overwrite.
+```
+cp -r node_modules/@seasketch/geoprocessing/dist/base-project/src/util/clipToGeography.ts src/util/clipToGeography.ts
```

Comment:
The cp command fails due to no `src/util` folder in the project, so I had to create the folder first. Not sure if this folder is created in template/default projects

Reply:
update guide to cover